### PR TITLE
Outofmemoryfix

### PR DESCRIPTION
**Pull request by kthare10**
https://github.com/RENCI-NRIG/exoimageservice/issues/3

### CODE REVIEW

File: src/main/java/org/renci/exogeni/image/service/oscontroller/OsImageController.java [hunk 1]
```diff
@@ -176,7 +182,65 @@ private String auth(String region) throws Exception {
...
+    public Integer getImageDetails(String imageId) throws Exception {
+
+        if (region == null || imageId == null) {
+            throw new ImageServiceException("Failed to construct image request; invalid input params");
+        }
+        try {
+            String token = auth(region);
+            if (token != null) {
+                HttpHeaders headers = new HttpHeaders();
+                System.out.println("Setting token " + token);
```
Comment (ibaldin) [inline src/main/java/org/renci/exogeni/image/service/oscontroller/OsImageController.java]: Maybe get rid or replace with logger call?

**Review by ibaldin — APPROVED** [review]
I didn't quite follow all the logic - i didn't see the part that uses openstack CLI to retrieve the image and save it locally. Is this part of it?

**Comment by ibaldin** [comment]
Can you maybe elaborate on the issue what the fix is - I didn't quite understand what this does beyond adding another call to the API. I thought you talked about using OpenStack CLI to retrieve the image and save it, but perhaps this is a different solution?

File: src/main/java/org/renci/exogeni/image/service/oscontroller/OsImageController.java (hunk 1 above)
Comment (kthare10) [reply src/main/java/org/renci/exogeni/image/service/oscontroller/OsImageController.java]: ok, will change to logger statement, I didn't have to use the command line tool, instead i could pass Range in the Header to download parts of the file. I have configured the python client to by default to download in parts of 50MB increments. If RANGE is not specified in the header the whole file is downloaded at once.

**Comment by ibaldin** [comment]
Oh I think I see it in the client - you made the client smarter to ask for chunks of the image.

**Comment by kthare10** [comment]
Updated the issue with the details for the fix.

**Pull request merged**